src/pages/StocktakesPage.js

```diff
@@ -54,95 +42,111 @@ export const StocktakesPage = ({ routeName, currentUser, dispatch: reduxDispatch
+  useNavigationFocus(callback, navigation);
+
+  const onRowPress = useCallback(stocktake => reduxDispatch(gotoStocktakeEditPage(stocktake)), []);
+  const onFilterData = value => dispatch(PageActions.filterData(value));
+  const onCancelDelete = () => dispatch(PageActions.deselectAll());
+  const onConfirmDelete = () => dispatch(PageActions.deleteStocktakes());
+  const onCloseModal = () => dispatch(PageActions.closeModal());
+
+  const onNewStocktake = () => {
+    if (usingPrograms) return dispatch(PageActions.openModal(MODAL_KEYS.PROGRAM_STOCKTAKE));
+    return reduxDispatch(gotoStocktakeManagePage({ stocktakeName: '' }));
+  };
 
-  const getAction = useCallback((colKey, propName) => {
+  const getAction = (colKey, propName) => {
     switch (colKey) {
       case 'remove':
-        if (propName === 'onCheckAction') return selectRow;
-        return deselectRow;
+        if (propName === 'onCheckAction') return PageActions.selectRow;
+        return PageActions.deselectRow;
+      default:
+        return null;
+    }
+  };
+
+  const getModalOnSelect = () => {
+    switch (modalKey) {
+      case MODAL_KEYS.PROGRAM_STOCKTAKE:
+        return ({ stocktakeName, program }) => {
+          reduxDispatch(createStocktake({ program, stocktakeName, currentUser }));
+          onCloseModal();
+        };
       default:
         return null;
     }
-  });
+  };
 
   const renderRow = useCallback(
     listItem => {
       const { item, index } = listItem;
       const rowKey = keyExtractor(item);
-      const { row, alternateRow } = newDataTableStyles;
-
       return (
         <DataTableRow
           rowData={data[index]}
           rowState={dataState.get(rowKey)}
           rowKey={rowKey}
-          style={index % 2 === 0 ? alternateRow : row}
           columns={columns}
           dispatch={dispatch}
           getAction={getAction}
-          onPress={() => reduxDispatch(gotoStocktakeEditPage(item))}
+          onPress={onRowPress}
+          rowIndex={index}
         />
       );
     },
     [data, dataState]
   );
 
-  const newStocktake = () => {
-    if (usingPrograms) return dispatch(openBasicModal(PROGRAM_STOCKTAKE));
-    return reduxDispatch(gotoStocktakeManagePage({ stocktakeName: '' }));
-  };
+  const renderHeader = useCallback(
+    () => (
+      <DataTableHeaderRow
+        columns={columns}
+        dispatch={instantDebouncedDispatch}
+        sortAction={PageActions.sortData}
+        isAscending={isAscending}
+        sortBy={sortBy}
+      />
+    ),
+    [sortBy, isAscending]
+  );
 
-  const renderButtons = () => {
+  const PageButtons = useCallback(() => {
     const { verticalContainer, topButton } = globalStyles;
     return (
       <View style={verticalContainer}>
-        <PageButton
-          style={topButton}
-          text={buttonStrings.new_stocktake}
-          onPress={() => newStocktake()}
-        />
+        <PageButton style={topButton} text={buttonStrings.new_stocktake} onPress={onNewStocktake} />
```

Comment:
I know you don't like destructuring underscore variables, but suggestion anyway:

```
const { new_stocktake: newStocktakeText } = buttonStrings
<PageButton style={topButton} text={newStocktakeText} onPress={onNewStocktake} />
```